cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -676,6 +681,9 @@ func (a *StaticAutoscaler) deleteCreatedNodesWithErrors() bool {
 			klog.Warningf("Cannot determine nodeGroup for node %v; %v", id, err)
 			continue
 		}
+		if nodeGroup == nil || reflect.ValueOf(nodeGroup).IsNil() {
+			return false, fmt.Errorf("node %s has no known nodegroup", node.GetName())
+		}
```

Comment:
static_autoscaler will always returned when nodegroup is nil and CA can not go on until CloudProviderNodeInstancesCache is refresh, which may up to 2 min.
can we refresh CloudProviderNodeInstancesCache here to sync up with cloud provider?

@bpineau @mwielgus